P/Python/build_tarballs.jl

```diff
@@ -86,9 +86,7 @@ products = Product[
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency("Expat_jll", v"2.2.7"; compat="2.2.7"),
-    # Future versions of bzip2 should allow a more relaxed compat because the
-    # soname of the macOS library shouldn't change at every patch release.
-    Dependency("Bzip2_jll", v"1.0.6"; compat="=1.0.6"),
+    Dependency("Bzip2_jll", v"1.0.8"; compat="1.0.8"),
     Dependency("Libffi_jll", v"3.2.1"; compat="~3.2.1"),
```

Comment:
May also want to change the compat bounds of expat and libffi, they're old too: #3024